modules/strelka/1.0/strelka.smk

```diff
@@ -0,0 +1,213 @@
+    log:
+        stdout = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_configure.stdout.log",
+        stderr = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_configure.stderr.log"
+    params:
+        opts = op.switch_on_wildcard("seq_type", CFG["options"]["configure"]),
+    wildcard_constraints:
+        pair_status = "no_normal"
+    conda:
+        CFG["conda_envs"]["strelka"]
+    shell:
+        op.as_one_line("""
+        configureStrelkaGermlineWorkflow.py 
+        --bam={input.tumour_bam}
+        --referenceFasta={input.fasta}
+        --runDir=$(dirname {output.runwf})
+        {params.opts} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+
+# Example variant filtering rule (single-threaded; can be run on cluster head node)
+rule _strelka_run:
+    input:
+        runwf = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/runWorkflow.py"
+    output:
+        vcf_dir = directory(CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/")
+        #snp = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/somatic.snvs.vcf.gz",
+        #indels = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/somatic.indels.vcf.gz"
+    log:
+        stdout = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_run.stdout.log",
+        stderr = CFG["logs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/strelka_run.stderr.log"
+    params:
+        opts = CFG["options"]["strelka"]
+    conda:
+        CFG["conda_envs"]["strelka"]
+    threads:
+        CFG["threads"]["strelka"]
+    resources: 
+        mem_mb = CFG["mem_mb"]["strelka"]
+    shell:
+        op.as_one_line("""
+        {input.runwf} -j {threads} 
+        {params.opts} 
+        >{log.stdout} 2> {log.stderr}
+        """)
+
+
+rule _strelka_unzip:
+    input:
+        vcf_dir = rules._strelka_run.output.vcf_dir
+        #vcf = CFG["dirs"]["strelka"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/results/variants/{var_type}.vcf.gz"
+    output:
+        vcf = CFG["dirs"]["filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{var_type}.vcf"
+    shell:
+        "zcat {input.vcf_dir}/{wildcards.var_type}.vcf.gz > {output.vcf}"
+
+
+rule _strelka_filter:
+    input:
+        vcf = rules._strelka_unzip.output.vcf
+    output:
+        vcf = CFG["dirs"]["filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{var_type}.filt.vcf"
+    shell:
+        op.as_one_line("""
+        grep '^#' {input.vcf} > {output.vcf} && 
+        awk '{{ if ($7 == \"PASS\") {{ print $0 }} }}' {input.vcf} >> {output.vcf}
```

Comment:
You can simplify this command with:
```
awk 'BEGIN {FS=OFS="\t"} $0 ~ /^#/ || $7 == "PASS"'
```

If all you have is a conditional, you don't need to specify `print $0`.

Reply:
Thanks!